packages/dressed-react/src/rendering/callbacks.ts

```diff
@@ -0,0 +1,45 @@
+import { type MessageComponentInteraction, type ModalSubmitInteraction, patchInteraction } from "./interaction.ts";
+
+type Handler = ((i: MessageComponentInteraction) => unknown) | ((i: ModalSubmitInteraction) => unknown);
+
+export const handlers = new Map<string, Handler>();
+
+/** The pattern to export for handling `onClick`/`onSubmit` callbacks */
+export const pattern = "@dressed/react-handler-:handlerId{-:fallback}";
+
+/** Creates a function for handling callback interactions */
+// biome-ignore lint/complexity/noBannedTypes: Need an empty record
+export function setupCallbackHandler<T extends Record<string, Handler> = {}>(fallbacks: T = {} as T) {
+  for (const key in fallbacks) handlers.set(key, fallbacks[key]);
+  return Object.assign(
+    async (
+      interaction: MessageComponentInteraction | ModalSubmitInteraction,
+      args: { handlerId: string; fallback?: string },
+    ) => {
+      const handler = handlers.get(args.handlerId) ?? handlers.get(args.fallback ?? "default");
+      // @ts-expect-error
+      interaction = interaction.$patched === Symbol.for("@dressed/react") ? interaction : patchInteraction(interaction);
+      if (!handler) throw new Error("Unknown callback, try adding a default fallback handler");
+      await handler(interaction as never);
+      if (!interaction.history.length) interaction.deferUpdate();
+    },
+    {
+      fallbacks: Object.fromEntries(Object.keys(fallbacks).map((k) => [k, k])) as {
+        [K in keyof Omit<T, "default">]: K;
+      },
+    },
+  );
+}
+
+export function registerHandler(handler: Handler, fallback?: string, id = randId()) {
+  handlers.set(id, handler);
+  const fbText = fallback ? `-${fallback}` : "";
+  return { custom_id: `@dressed/react-handler-${id}${fbText}`, $registeredHandler: id };
+}
+
+function randId(length = 16) {
```

Comment:
Could maybe benefit from https://github.com/paralleldrive/cuid2 here if you're wanting something more collision resistant. May have to benchmark

Reply:
That's 100% fair, just an option *if* you're wanting to have fully collision proof. The likelihood of clashing as it is currently is small enough to not need something else.